buildSrc/src/main/kotlin/conventions/publishing.gradle.kts

```diff
@@ -141,7 +130,7 @@ tasks.register("publishSnapshots") {
     description = "Publishes snapshots to Sonatype"
 
     if (version.toString().endsWith("-SNAPSHOT")) {
-        dependsOn(tasks.withType<PublishToMavenRepository>())
+        dependsOn(tasks.named("publishToSonatype"))
```

Comment:
Curious why this change is needed.

I'm also not clear how it "knows" to use the snapshot repo.  Is it keying off the version string?